docs/reference/query-dsl/has-parent-query.asciidoc

```diff
@@ -1,93 +1,120 @@
-executed in the parent document space, which is specified by the parent
-type. This query returns child documents which associated parents have
-matched. For the rest `has_parent` query has the same options and works
-in the same manner as the `has_child` query.
+Returns child documents whose <<parent-join,joined>> parent document matches a
+provided query. You can create parent-child relationships between documents in
+the same index using a <<parent-join,join>> field mapping.
+
+[WARNING]
+====
+Because it performs a join, the `has_parent` query is slow compared to other queries.
+Its performance degrades as the number of matching parent documents increases.
+Each `has_parent` query in a search can increase query time significantly.
+====
+
+[[has-parent-query-ex-request]]
+==== Example request
+
+[[has-parent-index-setup]]
+===== Index setup
+To use the `has_parent` query, your index must include a <<parent-join,join>>
+field mapping. For example:
 
 [source,js]
---------------------------------------------------
-GET /_search
+----
+PUT /my_index
 {
-    "query": {
-        "has_parent" : {
-            "parent_type" : "blog",
-            "query" : {
-                "term" : {
-                    "tag" : "something"
+    "mappings": {
+        "properties" : {
+            "my-join-field" : {
+                "type" : "join",
+                "relations": {
+                    "parent": "child"
                 }
             }
         }
     }
 }
---------------------------------------------------
-// CONSOLE
 
-Note that the `has_parent` is a slow query compared to other queries in the
-query dsl due to the fact that it performs a join. The performance degrades
-as the number of matching parent documents increases. If you care about query
-performance you should not use this query. However if you do happen to use
-this query then use it as less as possible. Each `has_parent` query that gets
-added to a search request can increase query time significantly.
-
-[float]
-==== Scoring capabilities
+----
+// CONSOLE
+// TESTSETUP
 
-The `has_parent` also has scoring support. The default is `false` which
-ignores the score from the parent document. The score is in this
-case equal to the boost on the `has_parent` query (Defaults to 1). If
-the score is set to `true`, then the score of the matching parent
-document is aggregated into the child documents belonging to the
-matching parent document. The score mode can be specified with the
-`score` field inside the `has_parent` query:
+[[has-parent-query-ex-query]]
+===== Example query
 
 [source,js]
---------------------------------------------------
+----
 GET /_search
 {
     "query": {
         "has_parent" : {
-            "parent_type" : "blog",
-            "score" : true,
+            "parent_type" : "parent",
             "query" : {
-                "term" : {
-                    "tag" : "something"
-                }
+                "match_all" : {}
```

Comment:
I am not sure what is the purpose of this `has_parent` query with `"match_all" : {}` query inside?  Is it to retrieve all child docs (something that we warned against)?  May be a better example would be to have a more specific query.

Reply:
That's a great point! I update the index setup and query examples with 31c720a.

Instead of `match_all`, it now uses a `term` query similar to the previous example.